client/components/search-form.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+export default class SearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      location: '',
+      geolocation: null,
+      message: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.getGeolocation = this.getGeolocation.bind(this);
+  }
+
+  handleChange(event) {
+    if (event.target.value === '') this.setState({ message: '' });
+    if (this.state.geolocation) {
+      this.setState({
+        geolocation: null,
+        message: 'Device location removed.'
+      });
+    } else {
+      this.setState({ location: event.target.value });
+    }
+  }
```

Comment:
i could be wrong, but do we need the user's geolocation elsewhere in the app? should we lift the geolocation state up to App?

Reply:
Geolocation is only needed for the fetch to Yelp. The response has a "center" property which is the same value which will be used to center the Google Map in feature 3.